docs/get-started/how-to-get-strava-data-python.md

```diff
@@ -0,0 +1,391 @@
+
+`http://127.0.0.1:5000/authorization?state=&code=xxxxxxxxxxx&scope=read`.
+
+4. Copy the values in the URL between **code** and the **&** symbol. Above, the
+x's represent the values that you need to copy.
+
+```python
+# Open the url that you created above in a web browser
+webbrowser.open(url)
+print(
+    """You will see a URL that looks like this:
+    http://127.0.0.1:5000/authorization?state=&code=12323423423423423423423550&scope=read,activity:read_all,profile:read_all,read_all
+    Copy the values between code= and & in the URL that you see in the browser."""
+)
+# Using input allows you to copy the code into your Python console (or Jupyter Notebook)
+code = input("Please enter the code that you received: ")
+
+print(
+    f"Great! Your code is {code}\n"
+    "Next, I will exchange that code for a token.\n"
+    "I only have to do this once."
+)
+```
+
+You only need to get a code from Strava once.
+
+Next, you are ready to exchange the code for a token. After this, you can
+refresh your token as often as you need to to grab your data.
+
+```python
+token_response = client.exchange_code_for_token(
+    client_id=client_id, client_secret=client_secret, code=code
+)
+# Save token locally as a pickle
+with open(path_to_save, "wb") as f:
+    pickle.dump(token_response, f)
+
+print("Token saved - hooray!")
+
+# Access token and refresh tokens
+access_token = token_response["access_token"]
+refresh_token = token_response["refresh_token"]  # You'll need this in 6 hours
+```
+
+## Token refresh
+
+Strava provides you with an access token that is good for 6 hours. However, you can refresh the token as needed if you wish to grab
+more data in the future.
+
+Above you save the token as a pickle so you can access it in the future.
+
+```python
+client.get_athlete()  # Get current athlete details
+```
+
+If you want to get more data in the future / after your 6-hour window that
+the current token is good for, you can refresh the token using the
+`client.refresh_access_token()` method.
+
+```python
+# Open the token object that you saved locally earlier
+with open(token_path_pickle, "rb") as f:
+    tokens = pickle.load(f)
+
+refresh_response = client.refresh_access_token(
+    client_id=client_id,
+    client_secret=client_secret,
+    refresh_token=refresh_token,
+)
+
+# TODO: Again this seems weird. Why doesn't this all get updated when you refresh
+# above in that method?
+
+# Update your client object with the refreshed token values
+# TODO: this seems like extra work? method should do this
+client.access_token = refresh_response["access_token"]
+client.refresh_token = refresh_response["refresh_token"]
+client.token_expires_at = refresh_response["expires_at"]
```

Comment:
I'm going to try this again in 6 hours to see how refresh works now. i haven't looked at it closely in a while.